apis/container/v1alpha1/nodepool_types.go

```diff
@@ -0,0 +1,637 @@
+}
+
+// Assign the PrivateGKECluster field of the supplied resource.CanReference,
+// assumed to be a NodePool.
+func (v *GKEClusterURIReferencerForNodePool) Assign(res resource.CanReference, value string) error {
+	np, ok := res.(*NodePool)
+	if !ok {
+		return errors.New(errResourceIsNotNodePool)
+	}
+
+	np.Spec.ForProvider.Cluster = value
+	return nil
+}
+
+// NodePoolObservation is used to show the observed state of the GKE Node Pool
+// resource on GCP.
+type NodePoolObservation struct {
+	// Conditions: Which conditions caused the current node pool state.
+	Conditions []*v1beta1.StatusCondition `json:"conditions,omitempty"`
+
+	// InstanceGroupUrls: [Output only] The resource URLs of the [managed
+	// instance
+	// groups](/compute/docs/instance-groups/creating-groups-of-mana
+	// ged-instances)
+	// associated with this node pool.
+	InstanceGroupUrls []string `json:"instanceGroupUrls,omitempty"`
+
+	// PodIpv4CidrSize: [Output only] The pod CIDR block size per node in
+	// this node pool.
+	PodIpv4CidrSize int64 `json:"podIpv4CidrSize,omitempty"`
+
+	// Management: NodeManagement configuration for this NodePool.
+	Management *NodeManagementStatus `json:"management,omitempty"`
+
+	// SelfLink: [Output only] Server-defined URL for the resource.
+	SelfLink string `json:"selfLink,omitempty"`
+
+	// Status: [Output only] The status of the nodes in this pool instance.
+	//
+	// Possible values:
+	//   "STATUS_UNSPECIFIED" - Not set.
+	//   "PROVISIONING" - The PROVISIONING state indicates the node pool is
+	// being created.
+	//   "RUNNING" - The RUNNING state indicates the node pool has been
+	// created
+	// and is fully usable.
+	//   "RUNNING_WITH_ERROR" - The RUNNING_WITH_ERROR state indicates the
+	// node pool has been created
+	// and is partially usable. Some error state has occurred and
+	// some
+	// functionality may be impaired. Customer may need to reissue a
+	// request
+	// or trigger a new update.
+	//   "RECONCILING" - The RECONCILING state indicates that some work is
+	// actively being done on
+	// the node pool, such as upgrading node software. Details can
+	// be found in the `statusMessage` field.
+	//   "STOPPING" - The STOPPING state indicates the node pool is being
+	// deleted.
+	//   "ERROR" - The ERROR state indicates the node pool may be unusable.
+	// Details
+	// can be found in the `statusMessage` field.
+	Status string `json:"status,omitempty"`
+
+	// StatusMessage: [Output only] Additional information about the current
+	// status of this
+	// node pool instance, if available.
+	StatusMessage string `json:"statusMessage,omitempty"`
+}
+
+// NodePoolParameters define the desired state of a Google Kubernetes Engine
+// node pool.
+type NodePoolParameters struct {
+	// TODO(hasheddan): currently, the format is actually
+	// projects/projectID/locations/clusterLocation/clusters/clusterName. Should
+	// we use this or should we trim off project and then add it back using the
+	// referenced provider?
```

Comment:
If the API wants us to specify the full `projects/projectID/locations/clusterLocation/clusters/clusterName` then I think the most faithful, flexible, and unsurprising (to those who know the API) recreation of that as an API field would be to require the user to write `projects/projectID/locations/clusterLocation/clusters/clusterName`, even if it's a little redundant.